### PR TITLE
Improve README docs

### PR DESCRIPTION
**Pull request by andyczerwonka**
As I go through trying to register the demo workflows using `jflyte`, I'm trying to improve the docs to help others with the struggles I'm having.

- [x] point to the latest version
- [x] fix typo's
- [x] add a note around registering workflows if you're running against the demo cluster

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -62,7 +63,9 @@ FLYTE_STAGING_LOCATION=s3://my-s3-bucket
 FLYTE_PLATFORM_INSECURE=True
 ```
 
-Package and run:
+**Note**: If you're registering against [the local Demo Flyte Cluster](https://docs.flyte.org/en/latest/user_guide/environment_setup.html#create-a-local-demo-flyte-cluster), you'll need to adjust the ports to align with it.
+
```
Comment (andyczerwonka) [inline README.md:67]: [I ran into this problem.](https://flyte-org.slack.com/archives/CP2HDHKE1/p1720639439470809)

File: README.md [hunk 2]
```diff
@@ -62,7 +63,9 @@ FLYTE_STAGING_LOCATION=s3://my-s3-bucket
...
+Package and register:
```
Comment (andyczerwonka) [inline README.md:68]: It does not run the workflows, it simply registers them as far as I can tell.

File: README.md [hunk 3]
```diff
@@ -22,15 +22,16 @@
...
 - Missing user documentation
 - Project being tested, and collecting feedback
 - No guarantees of API stability
 
 To learn more about Flyte refer to:
 
- - [Flyte homepage](https://flyte.org)
- - [Flyte master repository](https://github.com/lyft/flyte)
+- [Flyte homepage](https://flyte.org)
+- [Flyte master repository](https://github.com/lyft/flyte)
```
Comment (davidmirror-ops) [inline README.md:34]: I guess the master repo is now: https://github.com/flyteorg/flyte :)

**Review by davidmirror-ops — APPROVED** [review]
Thank you!

**Review by andresgomezfrr — APPROVED** [review]

**Pull request merged**